lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -0,0 +1,98 @@
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Org.Apache.REEF.IMRU.OnREEF.MapInputWithControlMessage;
+using Org.Apache.REEF.Network.Group.Operators;
+using Org.Apache.REEF.Network.Group.Task;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.IMRUTasks
+{
+    /// <summary>
+    /// Hosts the IMRU UpdateTask in a REEF task
+    /// </summary>
+    /// <typeparam name="TMapInput">Map input</typeparam>
+    /// <typeparam name="TMapOutput">Map output</typeparam>
+    /// <typeparam name="TResult">Final result</typeparam>
+    public sealed class UpdateTaskHost<TMapInput, TMapOutput, TResult> : ITask
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(UpdateTaskHost<TMapInput, TMapOutput, TResult>));
+
+        private readonly IReduceReceiver<TMapOutput> _dataReceiver;
+        private readonly IBroadcastSender<MapInputWithControlMessage<TMapInput>> _dataAndControlMessageSender;
+        private readonly IUpdateFunction<TMapInput, TMapOutput, TResult> _updateTask;
+
+        /// <summary>
+        /// </summary>
+        /// <param name="updateTask">The UpdateTask hosted in this REEF Task.</param>
+        /// <param name="groupCommunicationsClient">Used to setup the communications.</param>
+        [Inject]
+        private UpdateTaskHost(
+            IUpdateFunction<TMapInput, TMapOutput, TResult> updateTask,
+            IGroupCommClient groupCommunicationsClient)
+        {
+            _updateTask = updateTask;
+            var cg = groupCommunicationsClient.GetCommunicationGroup(IMRUConstants.CommunicationGroupName);
+            _dataAndControlMessageSender = cg.GetBroadcastSender<MapInputWithControlMessage<TMapInput>>(IMRUConstants.BroadcastOperatorName);
+            _dataReceiver = cg.GetReduceReceiver<TMapOutput>(IMRUConstants.ReduceOperatorName);
+        }
+
+        /// <summary>
+        /// Performs IMRU iterations on update side
+        /// </summary>
+        /// <param name="memento"></param>
+        /// <returns></returns>
+        public byte[] Call(byte[] memento)
+        {
+            var updateResult = _updateTask.Initialize();
+            MapInputWithControlMessage<TMapInput> message =
+                new MapInputWithControlMessage<TMapInput>(MapControlMessage.AnotherRound);
+
+            while (updateResult.HasMapInput)
+            {
+                message.Message = updateResult.MapInput;
+                _dataAndControlMessageSender.Send(message);
+                updateResult = _updateTask.Update(_dataReceiver.Reduce());
+                if (updateResult.HasResult)
+                {
+                    // TODO: Emit output somewhere.
+                }
+            }
+
+            message.ControlMessage = MapControlMessage.Stop;
+            _dataAndControlMessageSender.Send(message);
+
+            if (updateResult.HasResult)
+            {
+                // TODO: Emit output somewhere.
```

Comment:
Please file a JIRA for this and change the TODO to the new format: `//TODO[REEF-XXX]: Emit ...`